OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -176,6 +189,18 @@ public void Repulse()
 			SetPosition(self, CenterPosition + FlyStep(speed, repulsionForce.Yaw.Facing));
 		}
 
+		public void DisableRepulsing()
```

Comment:
I made some changes in the last commit. I think your approach solved this issue.